liquid/liquid.go

```diff
@@ -0,0 +1,95 @@
+package liquid
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/json"
+	"errors"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"time"
+
+	"github.com/VictoriaMetrics/fastcache"
+	"github.com/breez/server/breez"
+)
+
+func checkSwapId(fc *fastcache.Cache, apiURL, swapID string, body []byte) error {
+	c := &http.Client{Timeout: 2 * time.Second}
+	r, err := c.Get(apiURL + swapID)
+	if err != nil {
+		return err
+	}
+	defer r.Body.Close()
+	if r.StatusCode != 200 {
+		return errors.New("bad swap id")
+	}
+	s := struct {
+		Status string `json:"status"`
+	}{}
+	err = json.NewDecoder(r.Body).Decode(&s)
+	if err != nil {
+		return err
+	}
+	if s.Status != "swap.created" {
+		return errors.New("bad status")
+	}
+	h := sha256.Sum256(body)
```

Comment:
Do we need to compare the body? Why not only check if swap_id exists? We are only going to use it for this endpoint.

Reply:
Because if the tx was not broadcasted, we want to be able to retry to broadcast the same tx